main/stm32bl.c

```diff
@@ -86,11 +114,46 @@ int stm32bl_go(uint32_t addr)
     if (wait_resp() != STM32BL_ACK) {
         return ESP_FAIL;
     }
-
     send_address(addr);
     return wait_resp();
 }
 
+static int stm32bl_run_raw_code(uint32_t target_address, const uint8_t *code, uint32_t code_size)
+{
+	uint32_t stack_le = 0x20002000;
+	uint32_t code_address_le = target_address + 8 + 1; // thumb mode address (!)
+	uint32_t length = code_size + 8;
+	uint8_t *mem, *pos;
+	uint32_t address;
+
+	/* Must be 32-bit aligned */
+	if (target_address & 0x3) {
+		ESP_LOGE(TAG, "Error: code address must be 4 byte aligned\n");
+		return ESP_FAIL;
+	}
+
+	mem = malloc(length);
```

Comment:
This is a comparably small buffer (something between 20 and 30 bytes if I didn't get that wrong). Can't we just put it onto the stack and avoid malloc?

Reply:
We could omit this function completely and put the reset code in the reset function instead. We most likely will never use the run raw code function anyway. If we keep it in case we want to run raw code at any point, I would keep the malloc here, it really is not a big deal on the esp32

Reply:
Ok, makes sense to keep it then... I'm somehow still a bit afraid of any malloc as most other MCUs I've been working with don't have that much speed + memory as this one :)